content/process/git-flow.md

```diff
@@ -1,3 +1,95 @@
+7. `git checkout -b <your branch>` to use the contents of the now clean `tmp` branch as your new local branch
+8. `git push --set-upstream=<remote> <your branch> --force` to overwrite the remote version of your branch with the clean one
+
+If there was a pull request that used your branch, it will now automatically be updated and no longer have any conflicts.
+
+??? hint "If you made a mistake"
+
+    If you made a mistake and overwrote your remote and local branches with one that does not work, no worries!
+    You can always use [`git reflog`](#git-reflog) to get back to the previous state of your branch, before you deleted anything.
+
+## Git Reflog
+
+Reference logs, or "reflogs", record when the tips of branches and other references were updated in the local repository.
+This is very useful for us, in order to revert changes that were made by mistake and to get back to the previous state of a modified branch.
+
+You can use `git reflog show` to look at your reference logs and find the hashes of the references you are interested in, and then simply `git checkout <hash>` to get back to their state. From there, you can use `git checkout -b <branch name>` to create a branch from that point in history.
+
+## Git Restore
+
+`git restore` is a relatively new command in `git` which allows you to restore files from any point in history and even from other branches.
+This can be very useful for example if you want a specific file from a specific commit on another branch, you don't need to `cherry-pick` the whole commit, but instead of you run `git restore --source=<other branch> -- /path/to/file`.
+
+## Clean Commits
+
+Ideally, we want to keep the commit history of our projects clean, in order to make it easy to navigate, to track and understand changes.
+On the master branch, each commit should correspond to a single PR, and within each PR, ideally each commit should correspond to a single change, in a way that reviewing a PR commit-by-commit should make sense.
+
+This is not simple however, since it means that if you go ahead and code huge amounts at once, you will then need to either:
+
+* Create multiple commits from these changes, by interactively adding the specific parts that belong together into one commit for each feature/component of the changes.
+* Create one huge commit for now, and split it later on by using `git rebase` and amending the commit history.
+
+### Interactive Add
+
+When using `git add -i`, you have access to an interface that lets you go through the changes on your local branch and choose whether to add them to staging or not, block-by-block.
+This means you do not need to add a whole file to your next commit, but instead can add only the parts of it that are relevant to that commit.
+
+### Amending Commit History
+
+The other solution is to use `git rebase -i` to amend the commit history.
+In order to split a commit into multiple parts, you need to specify the `edit` directive for that commit.
```

Comment:
What does it mean: "edit directive"?

Reply:
When you use the -i option in a rebase procedure, git will open a file in your editor with the list of commits that your branch adds upon the base branch. Next to each commit hash is a directive which you can change. By default commits are unchanged, but you can choose to delete, edit, amend, fix up and so on :)